### PR TITLE
Creates the SyndieReverseCard

### PR DESCRIPTION
**Pull request by jcatjmeow**
### Intent of your Pull Request

https://streamable.com/1i8ifu

Adds a new traitor item! The 'Bluespace Projectile Weapon Disrupter'. This stealthy item is hidden inside a normal-looking uno reverse card. But watch out! If you shoot someone who is holding this, they will get your gun and take no damage! You, on the other hand, will get the card. Very confusing!

PR is tested. The card does not work on turrets, emitters, mechs, or simplemobs. You can only steal real life actual guns that you can hold in your hand.

The idea behind this weapon is to apply some counterplay/tactics to firefights. If you are being shot at you can pull out the card to try and gain an edge in the fight. But you can't use it to turn around and murder the person who had the gun in the first place, because the item will just activate again and you'll swap again. For you, the best course of action is to pull out a melee weapon and charge.

If your opponent knows about this card, they can counter by using a melee weapon. Or, if they just got the card, they can counter by keeping the card and switching to another weapon. There is a lot of emergent gameplay and mindgames and counterplay and rock-paper-scissorsy action here and I think traitor could just a little more of that. It's especially useful against things that do a lot of damage all at once, like another traitor's revolver.

Price is currently 6tc. I figured it should be slightly less than just, a regular gun, cause you can't actually kill anyone with this. You can buy 3 before running out. I'm open to feedback about the price! Nukeops currently have access to it, but since nukies will usually have the better gun, it might not be a good idea to give this to your opponents. I dunno could be a funny gimmick. 

This only triggers if you are HOLDING it. And it will trigger on any projectile fired from a gun. So even foam force stuff, disablers, etc.

As for the meta side of things, before this is used, it is indistinguishable from a regular uno card. After it has been used at least once, the 'cloaking field' will 'fail' and expose the red and black flashy lights circuitry, to indicate that something is up. Antags who examine this will see an explanation of what it does.
#### Changelog

:cl:  
rscadd: Added a new traitor item: Bluespace Projectile Weapon Disrupter! 6TC.
/:cl:

### CODE REVIEW

**Comment by maxion12345** [comment]
>be me
>sec
>shoot traitor
>uno reverse
>traitor shoots me
>hold card
>swap again
>banned for using traitor item as security
>feelsbadman

File: code/game/objects/items/syndicateReverseCard.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	var/cardColor = pick ("Red", "Green", "Yellow", "Blue") //this randomizes which color reverse you get!
+	name = "[cardColor] Reverse"
+	icon_state = "sc_[cardColor] Reverse_uno"
+
+/obj/item/syndicateReverseCard/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!(attack_type == PROJECTILE_ATTACK))
+		return FALSE //this means the attack goes through
+	if(istype(hitby, /obj/item/projectile))
+		var/obj/item/projectile/P = hitby
+		if(P.firer && P.fired_from && (P.firer != P.fired_from)) //if the projectile comes from YOU, like your spit or some shit, you can't steal that bro. Also protects mechs
```
Comment (JamieD1) [inline code/game/objects/items/syndicateReverseCard.dm]:
```suggestion
		if(P?.firer && P.fired_from && (P.firer != P.fired_from)) //if the projectile comes from YOU, like your spit or some shit, you can't steal that bro. Also protects mechs
```
Prevents runtime

File: code/game/objects/items/syndicateReverseCard.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	//next, we move the gun to the user and the card to the firer
+	user.put_in_hands(gun)
+	firer.put_in_hands(src)
+	to_chat(user, "The [src] vanishes from your hands, and [gun] appears in them!")
+	to_chat(user, "<span class='warning'>[gun] vanishes from your hands, and a [src] appears in them!</span>")
+	used = TRUE
+
+/obj/item/syndicateReverseCard/examine(mob/user)
+	. = ..()
+	if(user.mind.special_role)
```
Comment (JamieD1) [inline code/game/objects/items/syndicateReverseCard.dm]:
```suggestion
	if(is_special_character(user))
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Looks good apart from the below:

Also nitpicks:
card.dm
line 13 - pick ( to pick(

**Comment by jcatjmeow** [comment]
The card now becomes very obviously syndie after it's been used at least once. Red and black, flashing lights, duct tape, etc. This was at Alexkar's discretion. Also uh, the image is composted from the X4 sprite and the bluespace crystal one and some wires, sooo >.> please be gentle

**Comment by Jarod1200** [comment]
can we make it say "no u" on it

**Comment by jcatjmeow** [comment]
> can we make it say "no u" on it

That might be a bit tooooo lowrp. I'm already pushing it with the uno

File: code/game/objects/items/syndicateReverseCard.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		cut_overlays()
+		add_overlay(image('icons/obj/items_and_weapons.dmi',icon_state = "Reverse_Overlay"))
+
+/obj/item/syndicateReverseCard/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!(attack_type == PROJECTILE_ATTACK))
+		return FALSE //this means the attack goes through
+	if(istype(hitby, /obj/item/projectile))
+		var/obj/item/projectile/P = hitby
+		if(P?.firer && P.fired_from && (P.firer != P.fired_from)) //if the projectile comes from YOU, like your spit or some shit, you can't steal that bro. Also protects mechs
+			if(ismachinery(P.firer) || isanimal(P.firer) || issilicon(P.firer)) //You can't switcharoo with turrets or simplemobs, or borgs
```
Comment (JamieD1) [inline code/game/objects/items/syndicateReverseCard.dm]: Could you not just do ishuman?

File: code/game/objects/items/syndicateReverseCard.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+				return FALSE
+			switcharoo(P.firer, owner, P.fired_from)
+			return TRUE //this means the attack is blocked
+	return FALSE
+
+/obj/item/syndicateReverseCard/proc/switcharoo(mob/firer, mob/user, gun) //this proc teleports the gun out of the firer's hands and into the user's. The firer gets the card.
```
Comment (JamieD1) [inline code/game/objects/items/syndicateReverseCard.dm]:
```suggestion
/obj/item/syndicateReverseCard/proc/switcharoo(mob/firer, mob/user, obj/item/gun/gun) //this proc teleports the gun out of the firer's hands and into the user's. The firer gets the card.
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Try these changes for me,
Also:
syndicateReserveCard.dm
line 21 ,icon -> , icon

File: code/game/objects/items/syndicateReverseCard.dm (hunk 3 above)
Comment (jcatjmeow) [reply code/game/objects/items/syndicateReverseCard.dm]: This is more secure. Covers all the bases

File: code/game/objects/items/syndicateReverseCard.dm (hunk 4 above)
Comment (jcatjmeow) [reply code/game/objects/items/syndicateReverseCard.dm]: if I'm doing this I'd rather change the var name cause like, gun/gun is confusing

File: code/game/objects/items/syndicateReverseCard.dm (hunk 3 above)
Comment (JamieD1) [reply code/game/objects/items/syndicateReverseCard.dm]: I would feel like just limiting it to humans would be more secure imo. Ill talk with alexkar.

Comment (jcatjmeow) [reply code/game/objects/items/syndicateReverseCard.dm]: yeah ur right

**Review by JamieD1 — APPROVED** [review]

File: code/game/objects/items/syndicateReverseCard.dm [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	..()
+	var/cardColor = pick ("Red", "Green", "Yellow", "Blue") //this randomizes which color reverse you get!
+	name = "[cardColor] Reverse"
+	icon_state = "sc_[cardColor] Reverse_uno"
+
+/obj/item/syndicateReverseCard/update_icon()
+	..()
+	if (used)
+		cut_overlays()
+		add_overlay(image('icons/obj/items_and_weapons.dmi', icon_state = "Reverse_Overlay"))
```
Comment (alexkar598) [inline code/game/objects/items/syndicateReverseCard.dm]: icon states should be in all lowercase whenever possible.

~~Also you have a rogue white pixel in the top left corner in the last 5 frames~~

File: code/game/objects/items/syndicateReverseCard.dm [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+		var/obj/item/projectile/P = hitby
+		if(P?.firer && P.fired_from && (P.firer != P.fired_from)) //if the projectile comes from YOU, like your spit or some shit, you can't steal that bro. Also protects mechs
+			if(iscarbon(P.firer)) //You can't switcharoo with turrets or simplemobs, or borgs
+				switcharoo(P.firer, owner, P.fired_from)
+				return TRUE //this means the attack is blocked
+	return FALSE
+
+/obj/item/syndicateReverseCard/proc/switcharoo(mob/firer, mob/user, obj/item/gun/target_gun) //this proc teleports the target_gun out of the firer's hands and into the user's. The firer gets the card.
+	//first, the sparks!
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
```
Comment (alexkar598) [inline code/game/objects/items/syndicateReverseCard.dm:36]: Optional but would be nice to also have sparks on the person being fired on

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/syndicateReverseCard.dm (hunk 5 above)
Comment (alexkar598) [reply code/game/objects/items/syndicateReverseCard.dm]: Ignore the second part

**Comment by jcatjmeow** [comment]
@alexkar598

**Comment by jcatjmeow** [comment]
Sprites incoming I will let yall know when it's proper done

**Review by Partheo — APPROVED** [review]
Thought there was a stray pixel but it's just icondiffbot being dumb. 
image

Sprite is good.

**Pull request merged**